src/tracer.cc

```diff
@@ -337,13 +340,12 @@ TraceManager::Trace::CaptureTimestamp(
           << "{\"name\":\"" << name << "\",\"ns\":" << timestamp_ns << "}]}";
     } else if (setting_->mode_ == TRACE_MODE_OPENTELEMETRY) {
 #ifndef _WIN32
-      otel_common::SystemTimestamp otel_timestamp{
-          (time_offset_ + std::chrono::nanoseconds{timestamp_ns})};
-      if (trace_span_ == nullptr) {
-        InitSpan(otel_timestamp);
-      }
-      trace_span_->AddEvent(
-          name, otel_timestamp);
+      auto root_span = opentelemetry::nostd::get<
```

Comment:
As root_span is being used at multiple places, does it make sense to convert into class variable within Trace?

Reply:
This way we will still need to store it in the OpenTelemetry context. We've discussed the task of exposing OpenTelemetry tracer and Context to user through python API, so that the user could create custom spans. Root span can be stored in / accessed from the OTel context.

I don't think that we should store the root span in multiple places, but I am open to discussion.